src/android/toga_android/widgets/base.py

```diff
@@ -1,18 +1,36 @@
 from ..libs.activity import MainActivity
 from ..libs.android_widgets import Gravity
 
+from rubicon.java.jni import java
+
 from toga.constants import CENTER, JUSTIFY, LEFT, RIGHT
 
 
+def _get_activity(_cache=[]):
+    """
+    Android Toga widgets need a reference to the current activity to pass it as `context` when creating
+    Android native widgets. This may be useful at any time, so we retain a global JNI ref.
+
+    :param _cache: List that is either empty or contains 1 item, the cached global JNI ref
+    """
+    if _cache:
+        return _cache[0]
```

Comment:
This is a neat trick - is there any particular advantage to doing it this way, vs a "traditional" `global` assignment?

Reply:
I like it more aesthetically because it creates fewer named variables, and because it's very slightly fewer lines of code, and because it doesn't trigger the autoformatter's desire to insert two newlines compared to the below.

```
_cache = None

def function():
    global _cache
    if _cache:
        return _cache
    ...
```

Either is fine with me, and I'll happily adjust this to the module-level global approach if you prefer it.